### ISSUE TITLE
When using 'strict-order' pihole querries Custom 2 before Custom 1

### ISSUE THREAD

**Issue by HarvsG** [pi-hole/docker-pi-hole#877]
### Versions

- Pi-hole:  v5.3.1
- AdminLTE:  5.5
- FTL:  5.8.1

### Platform

- OS and version:  Ubuntu 18.04

- Platform:  Docker

### Expected behavior
It would make more sense for Custom 1 to be queried before Custom 2
_A clear and concise description of what you expected to happen._

### Actual behavior / bug
The reverse happens
_A clear and concise description of what the bug is._

### Steps to reproduce

Set 2 custom DNS servers
in etc/pihole
`echo "strict-order" > 09-strict-order.conf` and restart pihole
Look at the logs and then all will be forwarded to Custom 2

**Comment by DL6ER** [comment]
Ah, `docker`. @PromoFaux do you know off-hand how `DNS1` and `DNS2` make it from environment variables to `01-pihole.conf` ?

**Comment by PromoFaux** [comment]
`DNS1` and `DNS2` are [deprecated](https://github.com/pi-hole/docker-pi-hole#deprecated-environment-variables).. you should be using the `PIHOLE_DNS_` variable described in the README. (The `DNS1/2` variables still work for now but they could be removed in a future release)

BUT, that all said I don't think this is to do with docker. 

You're adding the environment variables, they're going into `setupVars.conf` as `PIHOLE_DNS_1` and `PIHOLE_DNS_2` in that order. This sounds more like something in `dnsmasq` / `FTL` . I don't know how the `strict-order` thing is supposed to work

@DL6ER - batting the ball back to you 🎾😉

**Comment by DL6ER** [comment]
I did some tests myself and everything seems to work, all my queries got forwarded to the first server I have defined - with one exception - that was a query from an impatient client that retried a query before the reply from the first server arrived. In this case, `dnsmasq`/FTL chose to deviate from `strict-order` and decided it would better forward to the second server.

---

Back to your particular issue - I'll summarize what we have:

```
docker-compose.yaml
DNS1: 172.16.251.3#53
DNS2: 172.16.251.4#5053
```
becomes
```
/etc/pihole/setupVars.conf
PIHOLE_DNS_1=172.16.251.3#53
PIHOLE_DNS_1=172.16.251.4#5053
```
becomes
```
/etc/dnsmasq.d/01-pihole.conf
server=172.16.251.3#53
server=172.16.251.4#5053
```

However, FTL is still sending everything to `172.16.251.4#5053` even though this is the second DNS server and `strict-order` is given in another file that is always present in the container.

@HarvsG Is this correct?

Can your provide the output of the command
```
grep "using nameserver" /var/log/pihole.log
```
from inside your container?
